File: common/src/main/java/generations/gg/generations/core/generationscore/config/LegendKeys.java
package generations.gg.generations.core.generationscore.config;

import net.minecraft.resources.ResourceLocation;

import java.util.Set;

public class LegendKeys {
    public static SpeciesKey ARTICUNO = create("articuno");
    public static SpeciesKey GALARIAN_ARTICUNO = create("articuno", Set.of("galarian"));
    public static SpeciesKey ZAPDOS = create("zapdos");
    public static SpeciesKey GALARIAN_ZAPDOS = create("zapdos", Set.of("galarian"));
    public static SpeciesKey MOLTRES = create("moltres");
    public static SpeciesKey GALARIAN_MOLTRES = create("moltres", Set.of("galarian"));

    public static SpeciesKey MEWTWO = create("mewtwo");

    public static SpeciesKey SUICUNE = create("suicune");
    public static SpeciesKey ENTEI = create("entei");
    public static SpeciesKey RAIKOU = create("raikou");

    public static SpeciesKey LUGIA = create("lugia");
    public static SpeciesKey HO_OH = create("hooh");
    public static SpeciesKey CELEBI = create("celebi");
    public static SpeciesKey REGIROCK = create("regirock");
    public static SpeciesKey REGICE = create("regice");
    public static SpeciesKey REGISTEEL = create("registeel");
    public static SpeciesKey REGIELEKI = create("regieleki");
    public static SpeciesKey REGIDRAGO = create("regidrago");

    public static SpeciesKey LATIAS = create("latias");
    public static SpeciesKey LATIOS = create("latias");

    public static SpeciesKey KYORGRE = create("kyorgre");
    public static SpeciesKey GROUDON = create("groudon");
    public static SpeciesKey RAYQUAZA = create("rayquaza");
    public static SpeciesKey DEOXYS = create("deoxys");

    public static SpeciesKey UXIE = create("uxie");
    public static SpeciesKey AZELF = create("azelf");
    public static SpeciesKey MESPRIT = create("mesprit");

    public static SpeciesKey DIALGA = create("dialga");
    public static SpeciesKey PALKIA = create("palkia");
    public static SpeciesKey GIRATINA = create("giratina");
    public static SpeciesKey HEATRAN = create("heatran");
    public static SpeciesKey REGIGIGAS = create("regigigas");
    public static SpeciesKey CRESSELIA = create("cresselia");
    public static SpeciesKey DARKRAI = create("darkrai");
    public static SpeciesKey PHIONE = create("phione");
    public static SpeciesKey MANAPHY = create("manaphy");
    public static SpeciesKey TORNADUS = create("tornadus");
    public static SpeciesKey THUNDURUS = create("thundurus");
    public static SpeciesKey LANDORUS = create("landorus");
    public static SpeciesKey ENAMORUS = create("enamorus");
    public static SpeciesKey RESHIRAM = create("reshiram");
    public static SpeciesKey ZEKROM = create("zekrom");
    public static SpeciesKey KYUREM = create("kyurem");
    public static SpeciesKey MELOETTA = create("meloetta");
    public static SpeciesKey HOOPA = create("hoopa");
    public static SpeciesKey TYPE_NULL = create("typenull");
    public static SpeciesKey TAPU_KOKO = create("tapukoko");
    public static SpeciesKey TAPU_LELE = create("tapulele");
    public static SpeciesKey TAPU_BULU = create("tapubulu");
    public static SpeciesKey TAPU_FINI = create("tapufini");
    public static SpeciesKey MAGEARNA = create("magearna");
    public static SpeciesKey MELMETAL = create("melmetal");
    public static SpeciesKey KUBFU = create("kubfu");
    public static SpeciesKey GLASTRIER = create("glastrier");
    public static SpeciesKey SPECTRIER = create("spectrier");
    public static SpeciesKey ZAMAZENTA = create("zamazenta");
    public static SpeciesKey ZACIAN = create("zacian");

    private static SpeciesKey create(String species) {
        return new SpeciesKey(new ResourceLocation("cobblemon", species));
    }

    private static SpeciesKey create(String species, Set<String> aspects) {
        return new SpeciesKey(new ResourceLocation("cobblemon", species), aspects);
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/config/LegendKeys.kt
package generations.gg.generations.core.generationscore.common.config

import net.minecraft.resources.ResourceLocation

object LegendKeys {
    @JvmField
    val WALKING_WAKE: SpeciesKey = create("walkingwake")
    @JvmField
    var ARTICUNO: SpeciesKey = create("articuno")
    @JvmField
    var GALARIAN_ARTICUNO: SpeciesKey = create("articuno", setOf("galarian"))
    @JvmField
    var ZAPDOS: SpeciesKey = create("zapdos")
    @JvmField
    var GALARIAN_ZAPDOS: SpeciesKey = create("zapdos", setOf("galarian"))
    @JvmField
    var MOLTRES: SpeciesKey = create("moltres")
    @JvmField
    var GALARIAN_MOLTRES: SpeciesKey = create("moltres", setOf("galarian"))

    @JvmField
    var MEWTWO: SpeciesKey = create("mewtwo")

    @JvmField
    var SUICUNE: SpeciesKey = create("suicune")
    @JvmField
    var ENTEI: SpeciesKey = create("entei")
    @JvmField
    var RAIKOU: SpeciesKey = create("raikou")

    @JvmField
    var LUGIA: SpeciesKey = create("lugia")
    @JvmField
    var HO_OH: SpeciesKey = create("hooh")
    @JvmField
    var CELEBI: SpeciesKey = create("celebi")
    @JvmField
    var REGIROCK: SpeciesKey = create("regirock")
    @JvmField
    var REGICE: SpeciesKey = create("regice")
    @JvmField
    var REGISTEEL: SpeciesKey = create("registeel")
    @JvmField
    var REGIELEKI: SpeciesKey = create("regieleki")
    @JvmField
    var REGIDRAGO: SpeciesKey = create("regidrago")

    @JvmField
    var LATIAS: SpeciesKey = create("latias")
    @JvmField
    var LATIOS: SpeciesKey = create("latias")

    @JvmField
    var KYOGRE: SpeciesKey = create("kyogre")
    @JvmField
    var GROUDON: SpeciesKey = create("groudon")
    @JvmField
    var RAYQUAZA: SpeciesKey = create("rayquaza")
    @JvmField
    var DEOXYS: SpeciesKey = create("deoxys")

    @JvmField
    var UXIE: SpeciesKey = create("uxie")
    @JvmField
    var AZELF: SpeciesKey = create("azelf")
    @JvmField
    var MESPRIT: SpeciesKey = create("mesprit")

    @JvmField
    var DIALGA: SpeciesKey = create("dialga")
    @JvmField
    var PALKIA: SpeciesKey = create("palkia")
    @JvmField
    var GIRATINA: SpeciesKey = create("giratina")
    @JvmField
    var HEATRAN: SpeciesKey = create("heatran")
    @JvmField
    var REGIGIGAS: SpeciesKey = create("regigigas")
    @JvmField
    var CRESSELIA: SpeciesKey = create("cresselia")
    @JvmField
    var DARKRAI: SpeciesKey = create("darkrai")
    var PHIONE: SpeciesKey = create("phione")
    @JvmField
    var MANAPHY: SpeciesKey = create("manaphy")
    @JvmField
    var TORNADUS: SpeciesKey = create("tornadus")
    @JvmField
    var THUNDURUS: SpeciesKey = create("thundurus")
    @JvmField
    var LANDORUS: SpeciesKey = create("landorus")
    @JvmField
    var ENAMORUS: SpeciesKey = create("enamorus")
    var RESHIRAM: SpeciesKey = create("reshiram")
    var ZEKROM: SpeciesKey = create("zekrom")
    var KYUREM: SpeciesKey = create("kyurem")
    @JvmField
    var GENESECT: SpeciesKey = create("genesect")
    @JvmField
    var MELOETTA: SpeciesKey = create("meloetta")
    @JvmField
    var HOOPA: SpeciesKey = create("hoopa")
    @JvmField
    var TYPE_NULL: SpeciesKey = create("typenull")
    @JvmField
    var TAPU_KOKO: SpeciesKey = create("tapukoko")
    @JvmField
    var TAPU_LELE: SpeciesKey = create("tapulele")
    @JvmField
    var TAPU_BULU: SpeciesKey = create("tapubulu")
    @JvmField
    var TAPU_FINI: SpeciesKey = create("tapufini")
    @JvmField
    var MAGEARNA: SpeciesKey = create("magearna")
    @JvmField
    var MELMETAL: SpeciesKey = create("melmetal")
    var KUBFU: SpeciesKey = create("kubfu")
    @JvmField
    var GLASTRIER: SpeciesKey = create("glastrier")
    @JvmField
    var SPECTRIER: SpeciesKey = create("spectrier")
    var ZAMAZENTA: SpeciesKey = create("zamazenta")
    var ZACIAN: SpeciesKey = create("zacian")
    @JvmField
    var ZYGARDE: SpeciesKey = create("zygarde")

    private fun create(species: String): SpeciesKey {
        return SpeciesKey(ResourceLocation.fromNamespaceAndPath("cobblemon", species))
    }

    private fun create(species: String, aspects: Set<String>): SpeciesKey {
        return SpeciesKey(ResourceLocation.fromNamespaceAndPath("cobblemon", species), aspects)
    }
}
